charset/charset.go

```diff
@@ -57,15 +57,18 @@ var charsetInfos = map[string]*Charset{
 	CharsetASCII:   {CharsetASCII, CollationASCII, make(map[string]*Collation), "US ASCII", 1},
 	CharsetLatin1:  {CharsetLatin1, CollationLatin1, make(map[string]*Collation), "Latin1", 1},
 	CharsetBin:     {CharsetBin, CollationBin, make(map[string]*Collation), "binary", 1},
+	CharsetGBK:     {CharsetGBK, CollationGBKChineseCI, make(map[string]*Collation), "GBK", 2},
```

Comment:
I have a config to control the parser can recognize the `gbk` or not, no need to add it here.
When this feature is GA, we will remove the config and add the do the things like this.